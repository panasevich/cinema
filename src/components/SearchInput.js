import React from 'react';
import PropTypes from 'prop-types';

const SearchInput = (props) => {
    const { onChange, onClick } = props;
    return (
        <React.Fragment>
            <input type="text" onChange={onChange} />
            <button type="button" onClick={onClick}>Search</button>
        </React.Fragment>
    );
};

SearchInput.propTypes = {
    onChange: PropTypes.func,
    onClick: PropTypes.func,
};

export default SearchInput;
