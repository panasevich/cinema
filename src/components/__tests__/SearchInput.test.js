import React from 'react';
import SearchInput from '../SearchInput';
import { shallow } from '../../helpers/enzyme';
import renderer from 'react-test-renderer';

describe('Search input tests', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<SearchInput onChange={() => {}} onClick={() => {}} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});