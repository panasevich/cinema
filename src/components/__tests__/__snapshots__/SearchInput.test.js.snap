// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search input tests renders correctly 1`] = `
Array [
  <input
    onChange={[Function]}
    type="text"
  />,
  <button
    onClick={[Function]}
    type="button"
  >
    Search
  </button>,
]
`;
